Sally7/RequestExecutor/ConcurrentRequestExecutor.cs

```diff
@@ -192,6 +192,7 @@ private class JobPool : IDisposable
         {
             private readonly Channel<int> _jobIdPool;
             private readonly Request[] _requests;
+            private volatile bool _disposed;
```

Comment:
Just a minor suggestion based on my preference and experience.

Instead of using the `volatile` modifier I prefer using explicit `Volatile.Read(ref)` / `Volatile.Write(ref, value)` as it's more obvious when reading the code. It's about visibility of the data, thus an important aspect (to prevent / hunt some bugs).

Reply:
I agree, marking the field `volatile` doesn't solve the visibility of the issue.